### PR TITLE
Add failing test for #6314  and PR #6356

### PR DESCRIPTION
**Pull request by astepin**
This is a failing test for issue #6314 in addition to fix in PR #6356 .

### CODE REVIEW

File: tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php [hunk 1]
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
...
+        $isbnMetadata = $this->generateIsbnEmbeddableFixture(['testEmbedded' => $testMetadata], 'prefix');
+        $isbnEntity = $this->newInstance($isbnMetadata);
+        $refClass = new \ReflectionClass($isbnEntity);
+
+        self::assertTrue($refClass->hasProperty('testEmbedded'));
+
+
+        $docComment = $refClass->getProperty('testEmbedded')->getDocComment();
+
+        $needle = sprintf('@Embedded(class="%s", columnPrefix=\'%s\')', $testMetadata->name,'prefix');
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php]: Fairly sure that we don't use single quotes on any annotations (see `columnPrefix`)

File: tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php [hunk 2]
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
...
+        self::assertContains($needle, $docComment);
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php]: I see [from the test runs](https://travis-ci.org/doctrine/doctrine2/jobs/240964836#L342) that this is the only failing bit: a fix should be easy to provide now :-)

File: tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php [hunk 3]
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
...
+
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php]: Remove this empty line

File: tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php [hunk 4]
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
...
+    public function testEmbeddedEntityWithColumnPrefix()
+    {
+        $testMetadata = $this->generateTestEmbeddableFixture();
+        $isbnMetadata = $this->generateIsbnEmbeddableFixture(['testEmbedded' => $testMetadata], 'prefix');
+        $isbnEntity = $this->newInstance($isbnMetadata);
+        $refClass = new \ReflectionClass($isbnEntity);
+
+        self::assertTrue($refClass->hasProperty('testEmbedded'));
+
+
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php]: Remove this empty line

File: tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php [hunk 5]
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
         return new $metadata->name;
     }
 
+    /**
+     * @group GH-6314
+     */
+    public function testEmbeddedEntityWithColumnPrefix()
+    {
+        $testMetadata = $this->generateTestEmbeddableFixture();
+        $isbnMetadata = $this->generateIsbnEmbeddableFixture(['testEmbedded' => $testMetadata], 'prefix');
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php]: Please use a random value for the prefix passed in here - makes it harder to escape checks

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by astepin** [comment]
I had to merge PR #6356 to my PR in order to solve another issue i've discovered. 
As described [here](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/tutorials/embeddables.html#column-prefixing), columnPrefix could be a string or a boolean.This has an impact on the generation of this annotation. I've fixed this and added appropriet tests.

**Comment by Ocramius** [comment]
LGTM :+1:

**Comment by Ocramius** [comment]
Thanks @astepin, sorry for the late feedback :-)

**Pull request merged**